### DISCUSSION TITLE
Is it possible to add the Extended MAC of child to the Child Table TLV of Network Diagnostics TLVs.

### DISCUSSION THREAD

**Discussion by jinpeng1989** [openthread/openthread#8300] [Q&A] [answered]
Thread protocol use the Route64 and the Child Timeout to draws the network topology for online devices. To display which physical device a node on the topology  corresponds to, application needs to provide the relationship between device identifier and rloc16. Because the rloc16 may change dynamically. The relationship will change dynamically change. This makes it difficult to implement the application. If add the Extended MAC of child to the Child Table, it will be much easier.

**Comment by jwhui** [comment, marked as answer]
[Thread 1.1.1](https://www.threadgroup.org/support#specifications) Section 10.11.4.4 defines the Child Table TLV for reporting Child Table entries. The current specification only includes the 9-bit Child ID for the identifier. While including the Extended Address would provide more information, that would grow each Child Table TLV by 8 bytes. Depending on the number of children attached, this can significantly increase the amount of information that needs to be communicated when reporting the child table.

Given that the Thread Group maintains the Thread Specification, I encourage you to bring the conversation to the Thread Group.
